### DISCUSSION TITLE
[Question] Automagically generate TS client?

### DISCUSSION THREAD

**Discussion by wedi** [tailflow/laravel-orion-ts#50] [Ideas]
Hi,
I'm evaluating Orion for a new project. It looks very promising so far. I am wondering if you have any plans to enable automatic generation of the API client that matches the actual models in Laravel without the need to manually specify the attributes?

It looks like most of the parts are there: an OpenAPI spec that knows the models and code that knows how to handle communication with the API.

This last piece would make Orion the ultimate one and only super power API package for Laravel. 🤓😄

**Comment by alexzarbn** [comment]
Hi @wedi,

As of now, this feature is not scheduled for any release, but it was hanging in the back of my mind for a while 🤔
While it is indeed possible to generate TypeScript models automatically, it only saves a couple of minutes off of the development time - not substantial, imho.

However, let's see, if it gets more traction, then I will schedule that for one of the releases 👌🏻

PRs are always welcome though :)
